### PR TITLE
Handle multiple errors

### PR DESCRIPTION
**Pull request by nduhamel**

### CODE REVIEW

File: errors/collector.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+func (c *Collector) HasErrors() bool {
+	if len(c.Errors) == 0 {
+		return false
+	}
+	return true
+}
+
+// Error implements error golang interface
+func (c *Collector) Error() string {
+	if len(c.Errors) == 0 {
```
Comment (gregdel) [inline errors/collector.go]: switch

File: errors/collector.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+		return fmt.Sprintf("No error")
+	} else if len(c.Errors) == 1 {
+		return c.Errors[0].Error()
+	}
+	str := fmt.Sprintf("Got %d errors:\\n", len(c.Errors))
```
Comment (gregdel) [inline errors/collector.go]: \\n ?

Comment (xunien) [inline errors/collector.go]: Why not use `HasErrors` ?

Comment (gregdel) [inline errors/collector.go]: It's the implementation of the golang built in erorr interface. 
I just don't understand why there is two "\"

Comment (xunien) [inline errors/collector.go]: Typo :)

Comment (nduhamel) [inline errors/collector.go]: \n is an error, a first I try to print all error's messages with \n between 

\\n is not a typo the first \ escape the second one in golang, the result string, wich is send to stdout is  \n

HasErrors is a remanent of previous code

File: errors/collector.go (hunk 1 above)
Comment (nduhamel) [inline errors/collector.go]: ok

File: lib/videostore.go [hunk 3]
```diff
@@ -263,7 +263,7 @@ func (vs *VideoStore) AddShowEpisode(ep *ShowEpisode) error {
 	if !exists(showNFOPath) {
 
 		show := NewShowFromEpisode(ep)
-		if err := show.GetDetails(); err != nil {
+		if err := show.GetDetails(vs.log); err != nil {
 			return err
```
Comment (PouuleT) [inline lib/videostore.go]: err might not be nil here, for example if one detailer fails, and the next one succeeds
You shouldn't return, you should log the error and check IsFatal() , like you did in the app

Comment (nduhamel) [inline lib/videostore.go]: yes, i miss it

File: server/downloader.go [hunk 4]
```diff
@@ -136,14 +134,14 @@ func (d *downloader) downloadMissingShows(wl *polochon.Wishlist) {
 	for _, wishedShow := range wl.Shows {
 		s := polochon.NewShow(d.config.Show)
 		s.ImdbID = wishedShow.ImdbID
-		s.SetLogger(d.log.WithField("imdbID", s.ImdbID))
+		log := d.log.WithField("imdbID", s.ImdbID)
 
-		if err := s.GetDetails(); err != nil {
+		if err := s.GetDetails(log); err != nil {
```
Comment (PouuleT) [inline server/downloader.go]: Same problem here, GetDetails now returns an error each time one of the detailers fails

File: lib/movie.go [hunk 5]
```diff
@@ -117,75 +111,94 @@ func readMovieNFO(r io.Reader, conf MovieConfig) (*Movie, error) {
...
-		err = t.GetTorrents(m, m.log)
+		err := t.GetTorrents(m, log)
 		if err == nil {
 			break
 		}
+		c.Push(errors.Wrap(err).Ctx("Torrenter", t.Name()))
 	}
-	return err
+
+	if c.HasErrors() {
```
Comment (PouuleT) [inline lib/movie.go]: Same change of behaviour here that might break the app

File: lib/movie.go [hunk 6]
```diff
@@ -117,75 +111,94 @@ func readMovieNFO(r io.Reader, conf MovieConfig) (*Movie, error) {
...
-		err = n.Notify(m, m.log)
+		err := n.Notify(m, log)
 		if err == nil {
 			break
 		}
+		c.Push(errors.Wrap(err).Ctx("Notifier", n.Name()))
+	}
 
-		m.log.Warnf("failed to send a notification from notifier: %q", err)
+	if c.HasErrors() {
```
Comment (PouuleT) [inline lib/movie.go]: And here too

File: lib/movie.go [hunk 7]
```diff
@@ -117,75 +111,94 @@ func readMovieNFO(r io.Reader, conf MovieConfig) (*Movie, error) {
...
 
 		if _, err := io.Copy(file, subtitle); err != nil {
-			return err
+			c.Push(errors.Wrap(err).Fatal())
+			return c
 		}
 	}
 
-	return err
+	if c.HasErrors() {
```
Comment (PouuleT) [inline lib/movie.go]: Here too

**Pull request merged**